lib/charms/traefik_k8s/v0/traefik_route.py

```diff
@@ -274,29 +312,50 @@ def get_static_config(self, relation: Relation) -> Optional[str]:
 
 
 class TraefikRouteRequirer(Object):
-    """Wrapper for the requirer side of traefik-route.
+    """Handles the requirer side of the traefik-route interface.
 
-    The traefik_route requirer will publish to the application databag an object like:
+    This class provides an API for publishing dynamic and static configurations
+    to the Traefik charm through the `traefik-route` relation. It does not perform
+    validation on the provided configurations, assuming that Traefik will handle
+    valid YAML-encoded data.
+
+    The application databag follows the structure:
+
+    ```json
     {
-        'config': <Traefik_config>
-        'static': <Traefik_config>  # optional
+        "config": "<Traefik dynamic config YAML>",
+        "static": "<Traefik static config YAML>",  # Optional, requires Traefik restart
+        "raw": "<bool>"  # Determines if Traefik should append TLS config for non HTTP routes
     }
+    ```
 
-    NB: TraefikRouteRequirer does no validation; it assumes that the
-    traefik-route-k8s charm will provide valid yaml-encoded config.
-    The TraefikRouteRequirer provides api to store this config in the
-    application databag.
     """
 
     on = TraefikRouteRequirerEvents()  # pyright: ignore
     _stored = StoredState()
 
-    def __init__(self, charm: CharmBase, relation: Relation, relation_name: str = "traefik-route"):
+    def __init__(
+        self,
+        charm: CharmBase,
+        relation: Relation,
+        relation_name: str = "traefik-route",
+        raw: Optional[bool] = False,
+    ):
         super(TraefikRouteRequirer, self).__init__(charm, relation_name)
         self._stored.set_default(external_host=None, scheme=None)
 
         self._charm = charm
         self._relation = relation
+        self._raw = raw
+
+        if self._raw:
+            log.warning(
+                "WARNING: 'raw' flag is set.\n"
+                "This will prevent Traefik from generating TLS routes for non-HTTP protocols,\n"
+                "even if a TLS provider is configured.\n"
+                "Charm authors should only use this option if they fully understand the implications\n"
+                "and intentionally do not require the appended TLS routes."
+            )
```

Comment:
I would not include this warning. It will only show up if raw was intentionally set, so it will not really ever result in action.